lib/application.js

```diff
@@ -559,6 +560,63 @@ app.listen = function(){
   return server.listen.apply(server, arguments);
 };
 
+/*
+ * Set a tracer at the application level. Tracer will be activated when a
+ * tracing call will be fired by a response.
+ * The tracer function takes an option object as argument with the given
+ * attributes:
+ *
+ * - app: current application
+ * - res: response that fires the tracer
+ * - req: request related to above response
+ * - event: string describing the event to trace
+ * - date: when the tracing occurs
+ * - args: allow to freely add information
+ *
+ * @param {function} The tracer to set.
+ * @return {app} for chaining.
+ * @api public
+ */
+
+app.instrument = function(tracer){
+  if (tracer === undefined || typeof tracer !== 'function')
+    throw new TypeError('instrument expects a function');
+  this.tracers.push(tracer);
+  return this;
+};
+
+/*
+ * Call all tracers set on the application. Add context information:
+ * running app, request, response and date. The date is the when the tracing
+ * was fired (now).
+ *
+ * @param {Response} Response that fires the tracing event.
+ * @param {Event} The event to trace.
+ * @param {Array} Arguments to transmit to the tracker.
+ *
+ * @api private
+ */
+
+app._callTracers = function(res, event, args){
+  var date = new Date();
```

Comment:
I was merely wondering what the purpose was, and I think you answered it well enough.